libvirt/resource_libvirt_domain_test.go

```diff
@@ -523,6 +593,103 @@ func testAccCheckLibvirtDomainDestroy(s *terraform.State) error {
 	return nil
 }
 
+func testAccCheckLibvirtDomainIsRunning(n string, domain *libvirt.Domain) resource.TestCheckFunc {
```

Comment:
this  Helper Function can be refactored with the other function.

I am in favor with the generic refactoring

```golang
func testAccCheckLibvirtDomaiStatusEqual (n string, domain *libvirt.Domain, domainStatusExpected string) 
```